src/service/standard-actions-impl.js

```diff
@@ -198,6 +220,28 @@ export class StandardActions {
     );
   }
 
+  /**
+   * Handles the `toggleTheme` action.
+   *
+   * This action sets the `amp-dark-mode` class on the body element and stores the the preference for dark mode in localstorage.
+   */
+  handleToggleTheme() {
+    this.ampdoc.waitForBodyOpen().then((body) => {
+      const prefersDarkScheme = this.ampdoc.win.matchMedia(
+        '(prefers-color-scheme: dark)'
+      ).matches;
+      const isDarkMode = this.ampdoc.win.localStorage.getItem('amp-dark-mode');
```

Comment:
It seems this is duplicated with above. Can we extract a simple `prefersDarkMode` read function?

Reply:
Addressed! I have added a `prefersDarkMode` function